Src/Common/WebHeaderCollectionExtensions.cs

```diff
@@ -55,8 +55,8 @@ public static void SetNameValueHeaderValue(this NameValueCollection headers, str
             Debug.Assert(headerName != null, "headerName must not be null");
             Debug.Assert(keyName != null, "keyName must not be null");
 
-            IEnumerable<string> headerValue = GetHeaderValue(headers, headerName);
-            headers[headerName] = string.Join(", ", HeadersUtilities.UpdateHeaderWithKeyValue(headerValue, keyName, value));
+            IEnumerable<string> headerValue = headers.GetHeaderValue(headerName);
+            headers[headerName] = string.Join(",", HeadersUtilities.UpdateHeaderWithKeyValue(headerValue, keyName, value));
```

Comment:
let me rephrase my question, why do we do it in this pull request? Is this related to the W3C implementation, or it's some optimization we want to make?

Reply:
we do this in this pull request because it's an obvious, nice and easy thing to do and improves things around tracestate implementation. I agree we may do it in the other PR, but being so subtle. it does not worth the hassle from my point of view